### DISCUSSION TITLE
How to retain the original stacktrace of an exception propagading between processes in .NET Core (.NET 6+)

### DISCUSSION THREAD

**Discussion by alexey-leonovich** [dotnet/runtime#110103] [Q&A] [answered]
I want to propagate exceptions between .NET Core (.NET 6+) processes and retain the original stacktrace. Currently we use .NET Remoting for inter-process comunication.
We used `Exception.PrepForRemoting` in .NET FW 472, but this method doesn't exist in .NET Core anymore so exception stack is truncated.
`ExceptionDispatchInfo.Throw(Exception source)` is not a solution because `ExceptionDispatchInfo` can't be serialized.
What is recommended approach for today?
Thank you.

**Comment by huoyaoyuan** [comment]
> We used `Exception.PrepForRemoting` in .NET FW 472, but this method doesn't exist in .NET Core anymore so exception stack is truncated.
> `ExceptionDispatchInfo.Throw(Exception source)` is not a solution because `ExceptionDispatchInfo` can't be serialized.

Remoting or binary serialization are deprecated and no longer a thing. You must design your own mechanism to format the stack trace.
There should be no way to retain the stack trace as-if it's thrown in the same process.

**Comment by stephentoub** [comment, marked as answer]
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.exceptionservices.exceptiondispatchinfo.setremotestacktrace?view=net-8.0#system-runtime-exceptionservices-exceptiondispatchinfo-setremotestacktrace(system-exception-system-string)

**Reply by alexey-leonovich** [reply]
Thank you. That could be handy.
